prow/resultstore/payload.go

```diff
@@ -121,14 +121,17 @@ func invocationAttributes(projectID string, pj *v1.ProwJob) *resultstore.Invocat
 
 func descriptionFromLabels(labels map[string]string) string {
 	jt := labels[kube.ProwJobTypeLabel]
-	repo := labels[kube.RepoLabel]
-	pull := labels[kube.PullLabel]
-	if ps := labels[kube.GerritPatchset]; ps != "" {
-		pull += "/" + ps
+	parts := []string{
+		labels[kube.RepoLabel],
 	}
-	buildID := labels[kube.ProwBuildIDLabel]
-	job := labels[kube.ContextAnnotation]
-	return fmt.Sprintf("%s for %s/%s/%s/%s", jt, repo, pull, buildID, job)
+	if pull := labels[kube.PullLabel]; pull != "" {
+		parts = append(parts, pull)
+		if ps := labels[kube.GerritPatchset]; ps != "" {
+			parts = append(parts, ps)
+		}
+	}
+	parts = append(parts, labels[kube.ProwBuildIDLabel], labels[kube.ProwJobAnnotation])
+	return fmt.Sprintf("%s for %s", jt, strings.Join(parts, "/"))
```

Comment:
QQ: Any reason why we can't just marshal into JSON or something more "standardized" here? Is it for human consumption?

Also why not just keep the label information for each of these values? Something like "job-type: <job-type>, RepoLabel: <RepoLabel>, PullLabel: <PullLabel>"? I think it's usually prudent to avoid lossy information transformations and I don't see a good reason why it's OK for this to be lossy. If there is a good reason, please add a comment explaining as such.

If we always had N number of "parts", where N was a fixed constant number, then I wouldn't worry  too much about this. But it appears to me that N can vary, and so the word coming after the first "/" may not always be the same type as in another run of this function. This could potentially mislead users/developers, hence my concern.

Reply:
Yes, the Description is the most prominent element in the ResultStore UI and intended for human consumption. Repo, Pull, etc. are all structured properties included in the UI under Invocation Details.

I offered to cjwagner that I could change Description so at least the optional parts were at the end. He noted that some jobs have very long names which make that less appealing and we don't want to bike shed, and perhaps templating descriptions is a better approach long term.